modules/netweaver/main.tf

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+terraform {
+  required_version = "~> 0.11.0"
+}
+
+locals {
+  device_name_1 = "${var.instance_name}-${var.device_1}"
+  device_name_2 = "${var.instance_name}-${var.device_2}"
+  device_name_3 = "${var.instance_name}-${var.device_3}"
+
+  access_config = {
+    "0" = []
+    "1" = [{}]
+  }
+}
+
+resource "google_compute_disk" "gcp_nw_pd_0" {
+  project = "${var.project_id}"
+  name    = "${var.instance_name}-nw-0"
+  type    = "${var.disk_type}"
+  zone    = "${var.zone}"
+  size    = "${var.usr_sap_size}"
+}
+
+resource "google_compute_disk" "gcp_nw_pd_1" {
+  project = "${var.project_id}"
+  name    = "${var.instance_name}-nw-1"
+  type    = "${var.disk_type}"
+  zone    = "${var.zone}"
+  size    = "${var.sap_mnt_size}"
+}
+
+resource "google_compute_disk" "gcp_nw_pd_2" {
+  project = "${var.project_id}"
+  name    = "${var.instance_name}-nw-2"
+  type    = "${var.disk_type}"
+  zone    = "${var.zone}"
+  size    = "${var.swap_size}"
+}
+
+resource "google_compute_attached_disk" "gcp_nw_attached_pd_0" {
+  project     = "${var.project_id}"
+  count       = "${var.usr_sap_size > 0 ? 1 : 0}"
```

Comment:
As I mentioned earlier, this logic must also be applied to the `google_compute_disk` resources. It would be wasteful for users to create 500 GB disks (when size = `0`) if they don't intend to attach them to the compute instance.

Reply:
made code changes for the disk too .